test/sym/assume.wat

```diff
@@ -4,18 +4,21 @@
   (import "symbolic" "assume_positive_i32" (func $positive_i32 (param i32)))
   (import "symbolic" "assert" (func $assert_i32 (param i32)))
 
-  (func $start
-    (local $x i32)
-    (local $y i32)
+  (func $start (local $x i32) (local $y i32)
+
     (local.set $x (call $i32_symbol))
     (local.set $y (call $i32_symbol))
+
     (call $positive_i32 (local.get $x)) ;; x >= 0
-    (call $assume_i32 (i32.gt_s (local.get $y) (i32.const 0))) ;; (y > 0) > 0
-    (if (i32.le_s (i32.add (local.get $x) (local.get $y)) (i32.const 0)) ;; x + y <= 0
-      (then unreachable))   ;; possible if i32 overflow
-    ;; Alternatively (x + y > 0)
+    (call $assume_i32 (i32.gt_s (local.get $y) (i32.const 0))) ;; y > 0
+
+    ;; if x + y <= 0
+    (if (i32.le_s (i32.add (local.get $x) (local.get $y)) (i32.const 0))
+      ;; possible if they overflow
+      (then unreachable))
+
+    ;; otherwise (x + y > 0)
     (call $assert_i32 (i32.gt_s (i32.add (local.get $x) (local.get $y)) (i32.const 0)))
```

Comment:
This is essentially testing the same thing. assert (x + y > 0) is the same as checking the satisfiability of (x + y <= 0)

Reply:
Yes I believe this tests was written to ensure we are adding the correct path condition when testing the condition in an `if then else`.

Reply:
I believe this was the same test I proposed to modify in https://github.com/OCamlPro/owi/pull/96#discussion_r1451038234. What I'm saying it that having:

```wast
  (if (i32.le_s (i32.add (local.get $x) (local.get $y)) (i32.const 0))
      (then unreachable))
```

Is equivalent to just writing: 

```wast
(call $assert_i32 (i32.gt_s (i32.add (local.get $x) (local.get $y)) (i32.const 0)))
```

And because this program always overflows the assert will never be executed. However, I agree that this test can also test what you're saying. But for that we would need to change `assume.t` to include `--no-stop-at-failure`:

```
symbolic extern module (assume and assert test):
  $ dune exec owi -- sym assume.wat --no-stop-at-failure
```

Reply:
> However, I agree that this test can also test what you're saying. But for that we would need to change assume.t to include --no-stop-at-failure:

Indeed, this was missing... I understand the confusion now. Will fix.